docs/guide/sending-email.md

```diff
@@ -0,0 +1,226 @@
+
+In this example, we are creating a custom piece type that implements an article. The `handlers()` function is being passed the `afterSave` server event. This event is emitted any time this custom module saves a new article. We are using this signal to send out an email to the site editors. The data coming from the page is passed into the `email.html` template through the `data` argument to add the title and blurb for the editors to review.
+
+::: note
+Note that in the Nunjucks template, the area is passed in through the `data` context. It isn't being passed in through the `data.piece` or `data.page` context as it would for a normal Nunjucks template view.
+:::
+
+## Triggering email from a route
+
+In addition to using `handlers()` to trigger email delivery, you can use `apiRoutes()`. This can be used to send an email when a user gets redirected to a selected page or sends a POST request, like a form submission. All of the same arguments passed when invoking the `email()` method from a module need to be supplied when using the method in a route. In this case, we need to create options and our data object from our `req.body`.
+
+### Example usage
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/widget-type',
+  options: {
+    lable: 'contact'
+  },
+  ...,
+  apiRoutes(self) {
+    return {
+      post: {
+        submit: async function (req) {
+          if (!req.body.email || !req.body.name) {
+            throw self.apos.error('invalid');
+          }
+          const email = self.apos.launder(req.body.email);
+          const options = {
+            from: 'admin@mysite.com',
+            to: email,
+            subject: 'Thanks for your submission!'
+          };
+          const data = {
+            name: self.apos.launder(req.body.name)
+          };
+          const template = 'email.html';
+          try {
+            await self.email(req, template, data, options);
+            return null;
+          } catch (err) {
+            self.apos.util.error('email notification error: ', err);
+          }
+        }
+      }
+    };
+  }
+};
+
+```
+
+<template v-slot:caption>
+/modules/article/index.js
+</template>
+
+</AposCodeBlock>
+
+In this case, the `email.html` template is the same as the one used for mail delivery from our `handlers()`.
+
+So, what is going on with this code? First, we are passing our `apiRoutes()` a `post` object. This contains the functions that should be used with a `POST` HTTP request. Each expected routes should get a separate function. In this case, we are passing the function through the `submit` property. This will monitor for a POST request to `https://www.mysite.com/api/v1/submit`. We could give the property a name prefixed with a slash to monitor that exact route - `https://www.mysite.com/submit`. This is useful for when the user is being redirected to a new page. See the [reference documentation](../reference/module-api/module-overview.html#naming-routes) for more details.
```

Comment:
I think you're suggesting an old school form submission on the front end here. That will run into CSRF protection problems unless excluded from CSRF. Given where new developers are starting out these days (JS, not traditional form submissions) I wouldn't worry about suggesting a redirect — I would think of this as an API to be programmatically invoked from frontend JavaScript code, perhaps using FormData.

Reply:
You should make the mapping between api route function names and routes explicit:

* `async submit(req)` -> `/api/v1/module-name/submit`
* `async incrementCounter(req)` -> `/api/v1/module-name/increment-counter` (note kebab case conversion)
* `'/submit': async (req)` -> `/submit`

Reply:
Updated, but I'm not sure it is what you want. I was trying to base it on your comments in the Linear ticket for PRO-3285.